daemon/cluster/executor/container/executor.go

```diff
@@ -143,9 +143,23 @@ func (e *executor) Configure(ctx context.Context, node *api.Node) error {
 	attachments := make(map[string]string)
 
 	for _, na := range node.Attachments {
+		if na == nil || na.Network == nil {
+			// this should not happen, but we got a panic here and don't have a
+			// good idea about what the underlying data structure looks like.
+			logrus.WithField("NetworkAttachment", fmt.Sprintf("%#v", na)).
+				Warnf("skipping nil node attachment entry")
+			continue
+		}
+
 		if na.Network.Spec.Ingress {
 			ingressNA = na
 		}
+
+		if len(na.Addresses) == 0 {
```

Comment:
Should this be put before the "if" above, to prevent `ingressNA` being set to an invalid (?) network, and the fallback code for the old Attachment not being used; https://github.com/moby/moby/blob/master/daemon/cluster/executor/container/executor.go#L152-L160